### PR TITLE
Fix login redirect

### LINKED ISSUE #2971: login doesn't redirect to original location
If someone sends me a link to something like `https://metabase.my-website.com/dash/22` and I'm not logged in, clicking the link will take me to the login page. After I log in it takes me to the activity feed instead of redirecting to `/dash/22` as I would expect.

### PR DESCRIPTION
**Pull request by tlrobinson**
Resolves #2971

Supersedes #3137

### CODE REVIEW

File: frontend/test/e2e/support/utils.js [hunk 1]
```diff
@@ -224,13 +227,32 @@ function jasmineMultipleSetupTeardown(fn) {
         const handlers = { beforeAll: [], beforeEach: [], afterEach: [], afterAll: [] }
         const originals = {};
 
+        let originalDescribe = global.describe;
```
Comment (kdoh) [inline frontend/test/e2e/support/utils.js]: This all looks fine but I'm not entirely sure what it's purpose is. Might be worth dropping a note as to its reason for being.

**Review by kdoh — COMMENTED** [review]
Left a note about comments but other than that LGTM.

File: frontend/test/e2e/auth/login.spec.js [hunk 2]
```diff
@@ -25,22 +25,22 @@ describeE2E("auth/login", () => {
...
+        it("should log you in", async () => {
             await driver.get(`${server.host}/`);
             await loginMetabase(driver, "bob@metabase.com", "12341234");
             await waitForUrl(driver, `${server.host}/`);
             const sessionCookie = await driver.manage().getCookie("metabase.SESSION_ID");
             sessionId = sessionCookie.value;
         });
 
-        xit ("should redirect you after logging in", async () => {
+        it("should redirect you after logging in", async () => {
```
Comment (salsakran) [inline frontend/test/e2e/auth/login.spec.js]: I'm a little confused here. Did we have the test for this *before* this PR? Did the test pass?

Comment (tlrobinson) [inline frontend/test/e2e/auth/login.spec.js]: TDD brah.

I can't remember when I originally added it but there was a separate PR for this awhile ago, but I merged the E2E stuff into it and it became a mess so I opened a new one.

Comment (salsakran) [inline frontend/test/e2e/auth/login.spec.js]: oh, sick. Are we going to be Agile too?

**Pull request merged**